### PR TITLE
Move the Checkbox validationHint outside of a tooltip

### LINKED ISSUE #1768: Make the Checkbox validationHint consistent with other form components
## Component to amend

`Checkbox`

## Visual

We're missing Figma specs for this component. Suggested changes will be reviewed by design

## Context

Currently, a Checkbox `validationHint` (description or warning/error message) is rendered as a tooltip. Beyond the fact that the current `Tooltip` component isn't accessible (see #1676), this is also visually inconsistent and has inherent accessibility issues (descriptions and status messages should not be concealed in a tooltip, they contain important information that helps users fill forms)

### PR DESCRIPTION
**Pull request by robinmetral**
Closes #1768 

## Purpose

Improve consistency and accessibility of the description and error messages for checkboxes.

## Approach and changes

- Move the Checkbox's `validationHint` (used for either a description or an error message) out of a tooltip and under the input (using the shared internal `FieldValidationHint` component)
- Update tests and stories (replace deprecated test utils, remove unnecessary snapshot+story)

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

File: packages/circuit-ui/components/Checkbox/Checkbox.stories.tsx [hunk 1]
```diff
@@ -91,17 +91,3 @@ Disabled.args = {
...
-      Apples
-    </CheckboxWithState>
-    <CheckboxWithState {...args} value="bananas" name="fruits">
-      Bananas
-    </CheckboxWithState>
-    <CheckboxWithState {...args} value="oranges" name="fruits">
-      Oranges
-    </CheckboxWithState>
-  </Fragment>
-);
```
Comment (connor-baer) [inline packages/circuit-ui/components/Checkbox/Checkbox.stories.tsx:107]: Why did you remove this story? This is a common use case, so I believe it's important to document how to implement it properly (i.e. associating the checkboxes using the `name` attribute).

File: packages/circuit-ui/components/Select/Select.tsx [hunk 2]
```diff
@@ -61,9 +61,9 @@ export interface SelectProps extends SelectHTMLAttributes<HTMLSelectElement> {
    */
   invalid?: boolean;
   /**
-   * Currently selected value. Matches the "value" property of
-   * the options objects. If value is falsy, Select will render
-   * the "placeholder" prop as currently selected.
+   * Currently selected value. Matches the "value" property of the options
+   * objects. If value is falsy, Select will render the "placeholder" prop as
+   * currently selected.
```
Comment (connor-baer) [inline packages/circuit-ui/components/Select/Select.tsx:66]: Do you have an editor plugin that does these line breaks automatically?

**Review by connor-baer — COMMENTED** [review]
Are you planning on connecting the validation hint to the input via `aria-describedby` in a separate PR?

File: packages/circuit-ui/components/Checkbox/Checkbox.stories.tsx (hunk 1 above)
Comment (robinmetral) [reply packages/circuit-ui/components/Checkbox/Checkbox.stories.tsx:107]: 🤦 I didn't think of the checkboxes-with-the-same-`name`-use-case.

Haven't seen it used a lot TBH (it's more common to have multiple checkboxes for multiple inputs, e.g. on auth.sumup.com for signup) but it's a valid point. Will add again

File: packages/circuit-ui/components/Select/Select.tsx (hunk 2 above)
Comment (robinmetral) [reply packages/circuit-ui/components/Select/Select.tsx:66]: Nope 🙈 just do it manually whenever I'm already tweaking descriptions in a file. A plugin (or prettier rule? 🥺) would be awesome

**Comment by robinmetral** [comment]
> Are you planning on connecting the validation hint to the input via aria-describedby in a separate PR?

In a separate PR 🙂 I'll do it for all form components.

Also a note that I'll make a separate PR to align the `label` prop across form components and deprecate any old behavior (normally using `childen`). Shouldn't be breaking (I want to make it opt-in), will be breaking in v7

Finally another note that this is under review by design, won't merge until I get a ✅ (I know I could do this on Chromatic but haven't onboarded the new joiners to the tool yet)

**Review by connor-baer — APPROVED** [review]

**Comment by robinmetral** [comment]
Got a ✅ from design, merging.
